sparrow-py/pysrc/sparrow_py/session.py

```diff
@@ -23,23 +58,51 @@ def __init__(self) -> None:
         self._session = _ffi.Session()
         self.tables = []
 
+
+    def validate_table(       
+        name: str,
+        time_column_name: str,
+        key_column_name: str,
+        schema: pa.Schema,
+        subsort_column_name: Optional[str] = None,
+        grouping_column_name: Optional[str] = None
+    ) -> None:
+        # Checking if the column exist. If they do not, it'll throw a KeyError.
+        key_field = schema.field(key_column_name)
+        if key_field.nullable:
+            raise KeyError(f"Key Column: {key_column_name} must be non-nullable")
+        time_field = schema.field(time_column_name)
+        if time_field.nullable:
+            raise KeyError(f"Time Column: {time_column_name} must be non-nullable")
+        
+        if subsort_column_name is not None:
+            subsort_field = schema.field(subsort_column_name)
+            if subsort_field.nullable:
+                raise KeyError(f"Subsort Column: {subsort_column_name} must be non-nullable")
+        if grouping_column_name is not None:
+            grouping_field = schema.field(grouping_column_name)
+            if grouping_field.nullable:
+                raise KeyError(f"Grouping Column: {grouping_column_name} must be non-nullable")
+
     def add_table(
         self,
         name: str,
         time_column_name: str,
         key_column_name: str,
         schema: pa.Schema,
         subsort_column_name: Optional[str] = None,
-        grouping_name: Optional[str] = None,
+        grouping_column_name: Optional[str] = None,
     ) -> Table:
+        Session.validate_table(name, time_column_name, key_column_name, schema, subsort_column_name=subsort_column_name, grouping_column_name=grouping_column_name)
```

Comment:
I don't think this should be on `Session`. It should be a class method on `Table` or a top-level method. We can discuss.